src/middleware/logger.rs

```diff
@@ -400,6 +421,7 @@ pub enum FormatText {
     RequestHeader(HeaderName),
     ResponseHeader(HeaderName),
     EnvironHeader(String),
+    CustomLog(String, Option<fn(req: &ServiceRequest) -> String>),
```

Comment:
So for the function, a common way around the Debug thing is a struct wrapping a function with a manual Debug impl.

I recommend using something like [this struct in actix-cors](https://docs.rs/actix-web/3.1.0/actix_web/struct.Resource.html#method.name):
```rust
struct CustomRequestFn {
    pub(crate) boxed_fn: Rc<dyn Fn(&HeaderValue, &RequestHead) -> bool>,}
```

Also using `Fn` over `fn` allows closures to capture from the environment (eg. a lazy instantiated regex). It's worth adding a test for this too.

Reply:
Awesome thanks for the heads up on this